src/execution_plan/optimizations/apply_join.c

```diff
@@ -182,38 +129,24 @@ static void _reduce_cp_to_hashjoin(ExecutionPlan *plan, OpBase *cp) {
 			// The entire Cartesian Product can be replaced with the join op.
 			ExecutionPlan_ReplaceOp(plan, cp, value_hash_join);
 			OpBase_Free(cp);
-			pending_filters_lookup_root = value_hash_join;
-			break;
+			// Try to popagate up the remaining filters.
```

Comment:
Let's expand this comment a little to emphasize that the optimization has been applied, and we'll now migrate all remaining filters and exit the loop.

Reply:
done